src/fs/storage.ts

```diff
@@ -72,10 +80,21 @@ export class InMemoryFSStorage implements FSStorage {
     this._data.clear();
     this._projectName = undefined;
   }
+  async markDirty(): Promise<void> {
+    this._dirty = true;
+  }
+  async clearDirty(): Promise<void> {
+    this._dirty = false;
+  }
+  async isDirty(): Promise<boolean> {
+    return this._dirty;
+  }
 }
 
-const fsPrefix = "fs/";
-const projectNameKey = "projectName";
+const fsFilesPrefix = "fs/files/";
+const fsMetadataPrefix = "fs/meta/";
+const projectNameKey = fsMetadataPrefix + "projectName";
+const dirtyKey = fsMetadataPrefix + "dirty";
```

Comment:
As discussed with Rob, we acknowledge this will break restore from session storage in the beta for users unlucky enough to rely on it for a page refresh before and after this PR going in.